core/flyout_horizontal.ts

```diff
@@ -82,7 +84,8 @@ export class HorizontalFlyout extends Flyout {
     if (!this.isVisible()) {
       return 0;
     }
-    const metricsManager = this.targetWorkspace!.getMetricsManager();
+    const targetWorkspace = this.targetWorkspace as WorkspaceSvg;
```

Comment:
do we even have flyouts in headless mode? i wonder if it makes more sense for the base flyout class to have the `targetWorkspace` use `WorkspaceSvg` instead of `Workspace`?

Reply:
It is a `WorkspaceSvg` in the base flyout class. This is about whether it's `null`, not about whether it's a `Workspace` or a `WorkspaceSvg`.

Reply:
IMO using `!` to assert non-null is just a special type of casting. If that's the only reason we're casting, I would prefer to just use `!` directly, as it's more clear about the intention, and tighter because it doesn't let you cast just whatever type into `WorkspaceSvg`. It only lets you remove `null` from the possible types. So I think using `!` is better than the general cast.

Therefore if we can't remove them, I would rather add them to the allowed style than change them to use looser casts.